CodingConventions.md

```diff
@@ -0,0 +1,307 @@
+Inbound data packages are appended to queue(s), and the vertex's internal worker threads process the inbound messages from the queue(s).
+In this design, there is an upper limit to the CPU resources an individual vertex can consume. A vertex with `k` go-routines as workers can
+at most keep `k` cores busy. The benefit of this desin is that the node can remain responsive even if one (or a few) of
+its data processing vertices are completely overwhelmed.
+
+In its entirety, a node is composed of many data flow vertices, that together form the node's data flow graph. The vertices
+within the node typically trust each other; while external messages are untrusted. Depending on the inputs it consumes, a vertex
+must carefully differentiate between trusted and untrusted inputs.
+
+⇒ **A vertex (data processing component within a node) implements the [`Component` interface](https://github.com/onflow/flow-go/blob/57f89d4e96259f08fe84163c91ecd32484401b45/module/component/component.go#L22)**.
+
+For example, [`VoteAggregator`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L29) is
+a data flow vertex within a consensus nodes, whose job it is to collect incoming votes from other consensus nodes.
+Hence, the corresponding function [`AddVote(vote *model.Vote)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L37)
+treats all inputs as untrusted. Furthermore, `VoteAggregator` consumes auxiliary inputs
+([`AddBlock(block *model.Proposal)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L45),
+[`InvalidBlock(block *model.Proposal)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L51),
+[`PruneUpToView(view uint64)`](https://github.com/onflow/flow-go/blob/3899d335d5a6ce3cbbb9fde4f7af780d6bec8c9a/consensus/hotstuff/vote_aggregator.go#L57))
+from the node's main consensus logic, which trusted.
+
+
+### Byzantine Inputs result in benign sentinel errors
+
+Any input that is coming from another node is untrusted and the vertex processing it should gracefully handle any arbitrarily malicious input.
+Under no circumstances should a byzantine input result in the vertex's internal state being corrupted.
+Per convention, failure case due to a byzantine inputs are represented by specifically-typed
+[sentinel errors](https://pkg.go.dev/errors#New) (for further details, see error handling section).
+
+Byzantine inputs are one particular case, where a function returns a 'benign error'. The critical property for an error
+to be benign is that the component returning it is still fully functional, despite encountering the error condition. All
+benign errors should be handled within the vertex.
```

Comment:
I think this section could be more explicit about the implementor weighing if an error is truly benign and having a high confidence in that. When that applies the benign error ensures no panic or fault in program flow.

Reply:
Added a sentence to this effect in https://github.com/onflow/flow-go/pull/2575/commits/6a5d51f431e91d89dd2b9efd570493dff2156667, let me know if you had specific wording in mind.

Reply:
Added a sentence to this effect in https://github.com/onflow/flow-go/pull/2575/commits/6a5d51f431e91d89dd2b9efd570493dff2156667, let me know if you had specific wording in mind.